google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudType.java

```diff
@@ -118,8 +118,8 @@ public void connect(@NotNull ConnectionCallback<AppEngineDeploymentConfiguration
 
       if (!Services.getLoginService().isLoggedIn()) {
         callback.errorOccurred(GctBundle.message("appengine.deployment.error.not.logged.in"));
-      } else if (CloudSdkUtil.isCloudSdkExecutable(
-          CloudSdkUtil.toExecutablePath(CloudSdkService.getInstance().getCloudSdkHomePath()))) {
+      } else if (CloudSdkUtil.isCloudSdkExecutable(CloudSdkUtil.toExecutablePath(
+          CloudSdkService.getInstance().getCloudSdkHomePath().toString()))) {
```

Comment:
The issue isn't that the path isn't valid, but rather it isn't set. That's the only way propertiesComponent.getValue(CLOUD_SDK_PROPERTY_KEY) can be set to null. Paths.get() actually doesn't perform any validation.
But we're still not doing any SDK path validation here, and the IDE is exploding when we set an invalid path. This would be solved by reusing CloudSdkPanel, which does the error checking.

@etanshaul should we move SDK checking to the CloudSdkService level, so that it doesn't let you create an SDK with an incorrect path?